### ISSUE TITLE
Replace QtWebKit with Qt WebEngine

### ISSUE THREAD

**Issue by agarny** [opencor/opencor#2325]
The last time we looked into it was in 2016 and we gave up on it (again!) because of printing and inspection not being possible or not being practical (see [here](https://github.com/opencor/opencor/issues/1142#issuecomment-259657621)). However, looking at the current version of Qt WebEngine, printing is now possible and inspection seems to be simpler (at least from the `webenginewidgets/printme` and `ebenginewidgets/simplebrowser` examples).

Now, until now, Qt WebKit was working fine for us, but it happens that while working on issue #2190 we came across a bug with WebKit itself. Basically, IFrames are rendered as blank pages (see [here](https://forum.qt.io/topic/113418/blank-page-as-a-result-of-using-iframes)) and this is a problem since now https://identifiers.org/ indirectly relies on IFrames (see [here](https://github.com/identifiers-org/identifiers-org.github.io/issues/92)).

So, if we want things to work with identifiers.org (which we do!) then we can't use QtWebKit...

**Comment by agarny** [comment]
Although a lot of work has been done to migrate to Qt WebEngine and that we are pretty close to being done with it (need to replace some of our DOM-related code with calls to JavaScript scripts), I am really not happy with Qt WebEngine. For instance:
 - It's a memory hog;
 - Its deployment proved ridiculous (the Qt documentation is not up to date in that regards!);
 - Saving a Web page to a PDF file is awfully slow;
 - To resize a WebViewer widget results in its contents taking time to be resized (while it's obviously instantaneous in Google Chrome, so why that delay in Qt WebEngine?!);
 - The execution of some JS scripts is asynchronous, so we had to make them synchronous so that our GUI workflow wouldn't be broken. However, this has resulted in some dirty hacks to get things to work roughly as we want. For instance, in the CellML Annotation view, if we select a CellML element that is annotate, we get the list of ontological terms, but it looks like things are initialised in the left of the window and then takes its final place after a split second.

The bottom line is that despite Qt WebEngine having now been around for several years, I still don't find it to be a worth replacement for QtWebKit. So, despite https://identifiers.org/ not working with QtWebKit, I prefer to stick to it since it results in a more robust and more professional looking OpenCOR.

As for QtWebKit, it's currently using the old WebKit, but will eventually be upgraded to the new WebKit which doesn't have problems with https://identifiers.org/ (see [here](https://forum.qt.io/topic/113418/blank-page-as-a-result-of-using-iframes/)). No ETA for that upgrade, but I really can't stand Qt WebEngine, so...

The codebase for our Qt WebEngine port can be found [here](https://github.com/opencor/opencor/files/4802236/opencor-issue2325.zip).
